src/clarity-angular/datagrid/datagrid-column-toggle.ts

```diff
@@ -74,46 +75,48 @@ export class DatagridColumnToggle implements OnInit, OnDestroy {
      * @type {Array}
      */
     public columns: DatagridHideableColumn[] = [];
-    public lastColumnshowing: boolean = false;
-    public allColumnsVisible: boolean;
+    // public lastColumnshowing: boolean = false;
+
+    public get allColumnsVisible(): boolean {
+        return this._allColumnsVisible;
+    }
+
+    public set allColumnsVisible(value: boolean) {
+        this._allColumnsVisible = value;
+    }
 
-    constructor( private hideableColumnService: HideableColumnService ) { }
+    constructor( public hideableColumnService: HideableColumnService ) { }
 
     ngOnInit() {
-        this.hideableColumnService.columnListChange.subscribe(( columnList ) => {
+
+        this._hideableColumnChangeSubscription =
+            this.hideableColumnService.columnListChange.subscribe(( columnList ) => {
             // Reset the list of columns
             this.columns.length = 0;
+            this.hideableColumnService.updateForLastVisibleColumn();
+            this.allColumnsVisible = this.hideableColumnService.checkForAllColumnsVisible;
 
             // Add only the hidden columns to the toggler.
             columnList.forEach(( col ) => {
                 if ( col ) {
-                    col.lastVisibleColumn = this.hideableColumnService.canHideNextColumn;
-                    this.allColumnsVisible = this.hideableColumnService.checkForAllColumnsVisible;
-
                     this.columns.push(col);
-                    this.hideableColumnService.updateForLastVisibleColumn();
-
-                    this._hideableColumnChangeSubscriptions.push(
-                        col.hiddenChange.subscribe(() => {
-                            col.lastVisibleColumn = this.hideableColumnService.canHideNextColumn;
-                            this.allColumnsVisible = this.hideableColumnService.checkForAllColumnsVisible;
-                        })
-                    );
                 }
             });
         });
     }
 
     ngOnDestroy() {
-        this._hideableColumnChangeSubscriptions.forEach(sub => sub.unsubscribe());
+        // this._hideableColumnChangeSubscription.unsubscribe();
     }
```

Comment:
Any reason this is commented? I'm guessing it was to test, but we should put it back.